substrate/frame/preimage/src/mock.rs

```diff
@@ -82,15 +82,18 @@ impl pallet_balances::Config for Test {
 	type ReserveIdentifier = [u8; 8];
 	type FreezeIdentifier = ();
 	type MaxFreezes = ConstU32<1>;
-	type RuntimeHoldReason = ();
```

Comment:
Why doesn't this work anymore?

Reply:
The `()` satisfies bounds `Parameter + Member + MaxEncodedLen + Copy` of `type RuntimeHoldReason: Parameter + Member + MaxEncodedLen + Copy + VariantCount;`, but the [`VariantCount` implementation](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/support/src/traits/misc.rs#L47-L49) returns `0`.

Before this PR, `type RuntimeHoldReason` and `MaxHolds` were disconnected, so you could have used `()` (VariantCount=0) with `MaxHolds = 2`, which it is kind of not clear, I guess.
But now, runtime developers does not need to think about setting correct values, they can provide whatever enum with `VariantCount` implemenetation. `()` basically means: "I don't support Holds feature".